aten/src/ATen/native/IndexingUtils.h

```diff
@@ -52,8 +52,8 @@ static C10_UNUSED void checkIndexTensorTypes(IOptTensorListRef indices) {
   for (const auto& tensor : indices) {
     if (tensor.has_value() && tensor->defined()) {
       auto scalarType = tensor->scalar_type();
-      if (scalarType != kLong && scalarType != kByte && scalarType != kBool) {
-          TORCH_CHECK_INDEX(false, "tensors used as indices must be long, byte or bool tensors");
+      if (scalarType != kLong && scalarType != kInt && scalarType != kByte && scalarType != kBool) {
```

Comment:
can you universally expand the types accepted by this function? It's called from many places, some of which (maybe) don't accept int32 indices.

Reply:
https://github.com/pytorch/pytorch/pull/86309 is more reasonable approach. Was trying to check if we can directly remove int constraint but looks like it fails even for CPU index ops.